src/functions/env.ts

```diff
@@ -191,49 +219,70 @@ function parseStrict(data: string): Record<string, string> {
  *
  * @return {Record<string, string>} Environment variables for the project.
  */
-export function load(options: {
+export function loadUserEnvs({
+  functionsSource,
+  projectId,
+  projectAlias,
+}: {
   functionsSource: string;
   projectId: string;
   projectAlias?: string;
 }): Record<string, string> {
-  const targetFiles = [".env"];
-  targetFiles.push(`.env.${options.projectId}`);
-  if (options.projectAlias && options.projectAlias.length) {
-    targetFiles.push(`.env.${options.projectAlias}`);
-  }
+  const envFiles = findEnvfiles(functionsSource, projectId, projectAlias);
 
-  const targetPaths = targetFiles
-    .map((f) => path.join(options.functionsSource, f))
-    .filter(fs.existsSync);
-
-  // Check if both .env.<project> and .env.<alias> exists.
-  if (targetPaths.some((p) => path.basename(p) === `.env.${options.projectId}`)) {
-    if (options.projectAlias && options.projectAlias.length) {
-      for (const p of targetPaths) {
-        if (path.basename(p) === `.env.${options.projectAlias}`) {
-          throw new FirebaseError(
-            `Can't have both .env.${options.projectId} and .env.${options.projectAlias}> files.`
-          );
-        }
-      }
+  // Disallow setting both .env.<projectId> and .env.<projectAlias>
+  if (projectAlias) {
+    if (envFiles.includes(`.env.${projectId}`) && envFiles.includes(`.env.${projectAlias}`)) {
+      throw new FirebaseError(`Can't have both .env.${projectId} and .env.${projectAlias}> files.`);
     }
   }
 
   let envs: Record<string, string> = {};
-  for (const targetPath of targetPaths) {
+  for (const f of envFiles) {
     try {
-      const data = fs.readFileSync(targetPath, "utf8");
+      const data = fs.readFileSync(path.join(functionsSource, f), "utf8");
       envs = { ...envs, ...parseStrict(data) };
     } catch (err) {
-      throw new FirebaseError(`Failed to load environment variables from ${targetPath}.`, {
+      throw new FirebaseError(`Failed to load environment variables from ${f}.`, {
         exit: 2,
         children: err.children?.length > 0 ? err.children : [err],
       });
     }
   }
-  logger.debug(
-    `Loaded environment variables ${JSON.stringify(envs)} from ${targetPaths.join(",")}.`
+  logBullet(
+    clc.cyan.bold("functions: ") +
+      `Loaded environment variables ${JSON.stringify(envs)} from ${envFiles.join(", ")}.`
```

Comment:
I don't know that we need to fully dump the list of environment variables. This could make the CLi very messy very quickly.